managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -145,7 +145,7 @@ private ManagedLedgerFactoryImpl(ZooKeeper zkc, ClientConfiguration bkClientConf
                 zkc, config, NullStatsLogger.INSTANCE);
     }
 
-    private ManagedLedgerFactoryImpl(ClientConfiguration clientConfiguration, String zkConnection, ManagedLedgerFactoryConfig config) throws Exception {
+    public ManagedLedgerFactoryImpl(ClientConfiguration clientConfiguration, String zkConnection, ManagedLedgerFactoryConfig config) throws Exception {
```

Comment:
Can we add support for creating ManagerLedgerFactoryImpl  with a Builder?
It is a code smell to add new public constructors.


WDYT?
Cc @codelipenghui

Reply:
Yes, we can create a builder for the `ManagerLedgerFactoryImpl`, we can do that in a separate PR I think.